packages/react/src/components/Dialog/Dialog.tsx

```diff
@@ -177,6 +177,21 @@ class Dialog extends AutoControlledComponent<ReactProps<DialogProps>, DialogStat
     },
   })
 
+  /**
+   * In case it generates an id, its uses will ensure that the id on the header
+   * and the value of aria-labelledBy on the dialog will be the same.
+   */
+  private getDefaultAriaLabelledBy() {
+    const { header } = this.props
+    if (this.props['aria-label'] || this.props['aria-labelledby'] || !header) {
+      return undefined
+    }
+    if (typeof header === 'string' || !header['id']) {
+      return _.uniqueId('dialog-header-')
```

Comment:
this will generate a different id on each render. it is not necessarily bad but maybe we can come up with a solution that would just use one for each render.

Slightly contradicting to this but more important - can we move the logic to the behavior and do similar logic for describedby? I just noticed in the issue that it explicitly says that we should not touch describedby - but that goes against what ARIA specifies: https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html

If we want to omit describedby, we can have an alternative behavior that just adds labelledby.

Reply:
> this will generate a different id on each render.

Hooks implementation: https://github.com/reach/reach-ui/blob/master/packages/auto-id/src/index.js

Reply:
I am confused. we want `same` id for each render or `different` id on each render?